### PR TITLE
Feature/16 Transaction Detail Page

### PR DESCRIPTION
**Pull request by ylembachar**
Adds the transaction detail page with search functionality.

### CODE REVIEW

File: frontend/src/routes.tsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ const AppRoutes = () => {
             <Routes>
                 <Route path="/system-overview" element={<System />} />
                 <Route path="/slot-overview" element={<Slot />} />
-                <Route path="/transaction" element={<Transaction />} />
+                <Route path="/transaction-detail" element={<Transaction />} />
```
Comment (fredo) [inline frontend/src/routes.tsx:14]: nitpick would call it `details`

File: frontend/cypress/components/SearchBar.cy.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                                <SearchBar
+                                    placeholder="Search by Txn Hash"
+                                    value={searchQuery}
+                                    onChange={handleSearchChange}
+                                    onSubmit={() => {}}
+                                />
+                            }
+                        />
+                        <Route
+                            path="/transaction-detail"
```
Comment (fredo) [inline frontend/cypress/components/SearchBar.cy.tsx:33]: same here

File: frontend/cypress/components/Transaction.cy.tsx [hunk 3]
```diff
@@ -1,58 +1,28 @@
...
-
-
         mount(
             <MemoryRouter initialEntries={[{ pathname: '/transaction-detail', state: transactionData }]}>
                 <Routes>
-                    <Route
-                        path="/transaction-detail"
-                        element={<Transaction />}
-                    />
+                    <Route path="/transaction-detail" element={<Transaction />} />
```
Comment (fredo) [inline frontend/cypress/components/Transaction.cy.tsx:12]: and here

File: frontend/cypress/components/Transaction.cy.tsx [hunk 4]
```diff
@@ -1,58 +1,28 @@
...
 
     it('renders a message when no transaction data is found', () => {
         mount(
             <MemoryRouter initialEntries={['/transaction-detail']}>
                 <Routes>
-                    <Route
-                        path="/transaction-detail"
-                        element={<Transaction />}
-                    />
+                    <Route path="/transaction-detail" element={<Transaction />} />
```
Comment (fredo) [inline frontend/cypress/components/Transaction.cy.tsx:25]: and here

File: frontend/src/routes.tsx (hunk 1 above)
Comment (ylembachar) [reply frontend/src/routes.tsx]: done for all :-)

**Review by fredo — APPROVED** [review]
looks good to me 👍

**Pull request merged**